src/main/java/Storage.java

```diff
@@ -0,0 +1,154 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+class Storage {
+
+    private File f;
+    private static final String path = "src/main/data/userTasks.txt";
+
+    public Storage() {
+        f = new File(path);
+    }
+
+    public ArrayList<Task> readFile() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(f);
+            while (sc.hasNext()) {
+                String sentence = sc.nextLine();
+                String[] splitted = sentence.split("\\|");
+                Task t;
+                if (splitted[1] != null) {
+                    if (splitted[1].startsWith("todo")) {
+                        t = addTodo(splitted[1]);
+                    } else if (splitted[1].startsWith("deadline")) {
+                        t = addDeadline(splitted[1]);
+                    } else if (splitted[1].startsWith("event")) {
+                        t = addEvent(splitted[1]);
+                    } else {
+                        t = new Task("dummy");
+                    }
```

Comment:
Could you avoid using arrowhead nesting if possible?

Reply:
Sure thing, thank you very much for your comments, really appreciate them. Will certainly take your comments into consideration for the next iterations! :smile: